### PR TITLE
2901 generate auth token

### PR DESCRIPTION
**Pull request by raftmsohani**
## Summary of Changes
Pull request closes #2901_

Added an endpoint to get token (that expires after X period of time) to call API endpoints

## How to Test
_List the steps to test the PR_
These steps are generic, please adjust as necessary.
```
cd tdrs-frontend && docker-compose -f docker-compose.yml -f docker-compose.local.yml up -d
cd tdrs-backend && docker-compose -f docker-compose.yml -f docker-compose.local.yml up -d 
```

1. Open http://localhost:3000/ and sign in.
1. Proceed with functional tests as described herein.
1. Test steps should be captured in the demo GIF(s) and/or screenshots below.
> *Demo GIF(s) and screenshots for testing procedure*

## Deliverables
_More details on how deliverables herein are assessed included [here](https://github.com/raft-tech/TANF-app/blob/develop/docs/How-We-Work/our-priorities-values-expectations.md#Deliverables)._

### [Deliverable 1: Accepted Features](https://github.com/raft-tech/TANF-app/blob/develop/docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-1-Accepted-Features)

Checklist of ACs:
- [x] token endpoint (`/v1/security/get-token`) only accessible to `OFA System Admin`
sysp1
non-sysp1_LI
- [x] token expires based on value of `TOKEN_EXPIRATION_HOURS`
- [x] API access using token works as expected
+ [x] **`lfrohlich`** and/or **`adpennington`**  confirmed that ACs are met.

### [Deliverable 2: Tested Code](https://github.com/raft-tech/TANF-app/blob/develop/docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-2-Tested-Code)

+ Are all areas of code introduced in this PR meaningfully tested?
  + [x] If this PR introduces backend code changes, are they meaningfully tested?
  + [ ] If this PR introduces frontend code changes, are they meaningfully tested?
+ Are code coverage minimums met?
  + [ ] Frontend coverage: [_insert coverage %_] (see `CodeCov Report` comment in PR)
  + [ ] Backend coverage: [_insert coverage %_] (see `CodeCov Report` comment in PR)

### [Deliverable 3: Properly Styled Code](https://github.com/raft-tech/TANF-app/blob/develop/docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-3-Properly-Styled-Code)

+ [x] Are backend code style checks passing on CircleCI?
+ [ ] Are frontend code style checks passing on CircleCI?
+ [x] Are code maintainability principles being followed?

### [Deliverable 4: Accessible](https://github.com/raft-tech/TANF-app/blob/develop/docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-4-Accessibility)

+ [ ] Does this PR complete the epic? 
+ [ ] Are links included to any other gov-approved PRs associated with epic?
+ [ ] Does PR include documentation for Raft's a11y review? 
+ [ ] Did automated and manual testing with `iamjolly` and `ttran-hub` using Accessibility Insights reveal any errors introduced in this PR?

### [Deliverable 5: Deployed](https://github.com/raft-tech/TANF-app/blob/develop/docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-5-Deployed)

+ [x] Was the code successfully deployed via automated CircleCI process to development on Cloud.gov?

### [Deliverable 6: Documented](https://github.com/raft-tech/TANF-app/blob/develop/docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-6-Code-documentation)

+ [x] Does this PR provide background for why coding decisions were made?
+ [ ] If this PR introduces backend code, is that code easy to understand and sufficiently documented, both inline and overall?
+ [ ] If this PR introduces frontend code, is that code easy to understand and sufficiently documented, both inline and overall?
+ [ ] If this PR introduces dependencies, are their licenses documented?
+ [x] Can reviewer explain and take ownership of these elements presented in this code review?

### [Deliverable 7: Secure](https://github.com/raft-tech/TANF-app/blob/develop/docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-7-Secure)

+ [x] Does the OWASP Scan pass on CircleCI?
+ [ ] Do manual code review and manual testing detect any new security issues?
+ [ ] If new issues detected, is investigation and/or remediation plan documented? 

### [Deliverable 8: User Research](https://github.com/raft-tech/TANF-app/blob/develop/docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-8-User-Research)

Research product(s) clearly articulate(s):
+ [ ] the purpose of the research
+ [ ] methods used to conduct the research 
+ [ ] who participated in the research
+ [ ] what was tested and how
+ [ ] impact of research on TDP
+ [ ] (_if applicable_) final design mockups produced for TDP development

### CODE REVIEW

File: tdrs-backend/docs/api/authentication.md
Comment (andrew-jameson) [inline tdrs-backend/docs/api/authentication.md]:
```suggestion
2. After user is logged in, user can grab a token at `/v1/security/get-token`
```

I assume it was just a typo. Does it need a trailing `/` as well?

File: tdrs-backend/docs/api/authentication.md
Comment (andrew-jameson) [inline tdrs-backend/docs/api/authentication.md]:
```suggestion
Note: the authentication token is available for 24 hours by default but this can be overridden using the `TOKEN_EXPIRATION_HOURS` environment variable.  
```

File: tdrs-backend/tdpservice/security/views.py
Comment (andrew-jameson) [inline tdrs-backend/tdpservice/security/views.py:16]:
```suggestion
    """Check if system admin can get a new token."""
```

**Review by andrew-jameson — APPROVED** [review]
Have some minor changes but otherwise looks good and works for me!

File: tdrs-backend/docs/api/authentication.md
Comment (raftmsohani) [reply tdrs-backend/docs/api/authentication.md]: Yes, and doesn't need the trailing `/`

File: tdrs-backend/tdpservice/security/utils.py
Comment (raftmsohani) [inline tdrs-backend/tdpservice/security/utils.py]: we should only log info here, don't have to raise exception.

**Review by elipe17 — CHANGES_REQUESTED** [review]
Gave async feedback to Mo about the exception blocking new token generation after expiration.

File: tdrs-backend/tdpservice/security/utils.py
Comment (raftmsohani) [reply tdrs-backend/tdpservice/security/utils.py]: Thanks @elipe17 for catching this 🥇

**Review by elipe17 — APPROVED** [review]
Works as advertised! Tested getting initial token and using it, letting token expire and verifying I couldnt query api anymore, and then getting a new token and checking I could query api.

File: tdrs-backend/tdpservice/security/views.py
Comment (jtimpe) [inline tdrs-backend/tdpservice/security/views.py]: should `access_request_status == 'APPROVED'` be included in this check?

**Review by jtimpe — APPROVED** [review]
Small question, everything else works great!

File: tdrs-backend/tdpservice/security/views.py
Comment (raftmsohani) [reply tdrs-backend/tdpservice/security/views.py]: You are correct. I will add this 💯

File: tdrs-backend/docs/api/authentication.md
Comment (ADPennington) [inline tdrs-backend/docs/api/authentication.md]: @raftmsohani ofa admin or ofa system admin? both?

Comment (raftmsohani) [reply tdrs-backend/docs/api/authentication.md]: It is only for OFA SYS ADMIN. Do you want ofa admin to have access too?

Comment (ADPennington) [reply tdrs-backend/docs/api/authentication.md]: ofa system admin is fine with me. just need to update this markdown with the relevant permission group.

Comment (raftmsohani) [reply tdrs-backend/docs/api/authentication.md]: Done!

File: tdrs-backend/docs/api/authentication.md
Comment (ADPennington) [inline tdrs-backend/docs/api/authentication.md:17]: @raftmsohani is this env var saved in circleci? or can this only be overridden with a codebase change?

Comment (raftmsohani) [reply tdrs-backend/docs/api/authentication.md:17]: @ADPennington This should be set to circle CI. I have not added the env var yet, was waiting for approval first.

**Comment by ADPennington** [comment]
@raftmsohani I'm not too sure if I'm missing something, but here are my results:

tokenp1

note:

- tried this after logging in via login.gov and via ams
- also got the same results via swagger (below)
tokenp2

**Comment by raftmsohani** [comment]
> @raftmsohani I'm not too sure if I'm missing something, but here are my results:
> 
> tokenp1
> 
> note:
> 
> * tried this after logging in via login.gov and via ams
> * also got the same results via swagger (below)
>   tokenp2

@ADPennington : The first time error above is because you have ```/``` at the end. the second error looks like you have added the environment variable as a string. If you want to add the env var for ```TOKEN_EXPIRATION_HOURS```

**Review by ADPennington — APPROVED** [review]
🥇

**Pull request merged**
